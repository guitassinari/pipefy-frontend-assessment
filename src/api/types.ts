export type Pipe = {
  name: string
  color: string
  cards_count: number
  id: number
}

export type Card = {
  title: string
}

export type CardConnetionEdge = {
  node: Card
  cursor: string
}