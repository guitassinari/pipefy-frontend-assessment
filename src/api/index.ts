import Queries from './queries'
import * as Types from './types'

export {
  Queries,
  Types
}