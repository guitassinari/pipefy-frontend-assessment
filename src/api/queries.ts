import { gql } from '@apollo/client'

const GET_ORGANIZATION = gql`
  query GetOrganization($id: ID!) {
    organization(id: $id) {
      id
      name
      pipes {
        id
        name
        color
        cards_count
      }
    }
  }
`

const GET_PIPE_CARDS = gql`
  query GetPipeCards($pipeId: ID!, $pageSize: Int, $startAtCursor: String) {
    cards(pipe_id: $pipeId, first: $pageSize, after: $startAtCursor) {
      pageInfo {
        endCursor
        hasNextPage
      }
      edges {
        cursor
        node {
          title
        }
      }
    }
  }
`

export default {
  GET_ORGANIZATION,
  GET_PIPE_CARDS
}