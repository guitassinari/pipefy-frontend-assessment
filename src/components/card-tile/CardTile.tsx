import React, { FC, HTMLAttributes } from 'react'
import { Types } from 'api'
import Tile from 'components/lib/tile'

interface CardTileProps {
  card: Types.Card,
}

const CardTile: FC<CardTileProps & HTMLAttributes<HTMLDivElement>> = ({ card, ...rest }) => (
  <Tile {...rest}>
    <div>{card.title}</div>
  </Tile>
)

export default CardTile