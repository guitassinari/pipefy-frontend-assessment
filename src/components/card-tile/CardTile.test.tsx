import React from 'react';
import { render } from '@testing-library/react'
import { Types } from 'api'
import CardTile from '.';

describe('CardTile', () => {
  describe('when receives a valid pipe', () => {
    const card: Types.Card = {
      title: 'a card'
    }

    it('matches the snapshot', () => {
      const component = render(<CardTile card={card} />)
  
      expect(component.asFragment()).toMatchSnapshot()
    })

    it('shows the card name', () => {
      const component = render(<CardTile card={card} />)
  
      expect(component.getByText(card.title)).toBeInTheDocument()
    })
  })
})
