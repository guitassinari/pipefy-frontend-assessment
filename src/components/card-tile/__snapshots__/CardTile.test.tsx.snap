// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTile when receives a valid pipe matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL kgCPtU"
  >
    <div>
      <div>
        a card
      </div>
    </div>
  </div>
</DocumentFragment>
`;
